rfcs/execute_script.md

```diff
@@ -0,0 +1,82 @@
+# RFC 89: `testdriver` Add an `execute_script` method.
+
+## Summary
+
+Add an `execute_script` method to testdriver that takes a js function,
+arguments to pass to the function, and the context id to execute it it
+in a remote context. If the function returns a promise we return the
+resolved value of the promise, otherwise we return the return value of
+the function.
+
+## Details
+
+Sometimes a test window wants to check the status of something in a
+remote window. It's possible to do this in a number of ways, for
+example by posting messages from the remote window to the test
+window. However in cross-origin cases messge passing on the js side
+can be difficult or impossible. In addition when tests are defined in
+the test window, it's useful to avoid putting complex logic in other
+browsing contexts because these won't have the same error handling
+properties as the test window.
+
+A simple way to solve these problems is to provide a mechansim to
+execute script in a different context. This isn't possible using
+content APIs alone, but is possible in WebDriver, and we can reuse
+that to define a `execute_script` function that works across contexts:
+
+```
+execute_script(fn, args, context)
+```
+
+`body` is a js function for the script to execute. To pass it to the
```

Comment:
References in `fn` are evaluated on `context`, while `args` are evaluated on the caller context, right?
e.g. 

```
execute_script(
    (arg) => [location.href, `${location.href}`, arg],
    [location.href, `${location.href}`],
    context)
```
would result in [the URL of `context`, the URL of `context`, the URL of the caller of `execute_script`, the URL of the caller of `execute_script`].

Reply:
Yes, that's right.

Reply:
what is "body" here? Should this be "fn"?